scripts/hypervisor-facts.py

```diff
@@ -5,88 +5,94 @@
+
     config['name'] = host
     # Ignore manually defined vms
     if any((d['name'] == host and 'org' in config and d['org'] == config['org']) for d in facts['xen_vman_vms']):
-      continue
+        continue
     if 'storage_type' in config:
-      storage_type = config['storage_type']
+        storage_type = config['storage_type']
     elif 'vm_facts_default_storage_type' in facts:
-      storage_type = facts['vm_facts_default_storage_type']
+        storage_type = facts['vm_facts_default_storage_type']
     else:
-      storage_type = 'filesystem'
+        storage_type = 'filesystem'
     # Set precise connection type for xen, depending on general storage_type
     if storage_type == 'blockdevice':
-      config['storage_type'] = 'iscsi'
+        config['storage_type'] = 'iscsi'
     else:
-      config['storage_type'] = 'nfs'
+        config['storage_type'] = 'nfs'
 
     # Copy description if necessary
     if 'description' not in config:
-      if 'description' in original_facts[host]:
-        config['description'] = original_facts[host]['description']
-      else:
-        failed_names['description'].append(host)
+        if 'description' in original_facts[host]:
+            config['description'] = original_facts[host]['description']
+        else:
+            failed_names['description'].append(host)
 
     # Copy interfaces if necessary
     if 'interfaces' not in config:
-      if 'interfaces' in original_facts[host]:
-        config['interfaces'] = original_facts[host]['interfaces']
-      else:
-        failed_names['interfaces'].append(host)
+        if 'interfaces' in original_facts[host]:
+            config['interfaces'] = original_facts[host]['interfaces']
+        else:
+            failed_names['interfaces'].append(host)
 
     # Set CIDR subnet mask
     for interface in config['interfaces']:
-      if 'ip' in interface and not re.search("/[0-9]+", interface['ip']):
-        interface['ip'] = interface['ip'] + cidr_suffix
+        if 'ip' in interface and not re.search("/[0-9]+", interface['ip']):
+            interface['ip'] = interface['ip'] + cidr_suffix
 
     # Remove unneeded filesystems attribute
     if 'filesystems' in config:
-      del config['filesystems']
-
-    facts['xen_vman_vms'].append(config)
-
-  # Return the result, consisting of the failed hosts and the extended hostvars for the current hypervisor
-  result = {'failed_hosts': failed_names, 'new_hostvars': facts}
-  return result
+        del config['filesystems']
 
+    if hypervisor_host == hypervisor_for_vm:
+        facts['xen_vman_vms'].append(config)
+    elif 'pull_hypervisor_from' in config and config['pull_hypervisor_from'] == hypervisor_host:
+        # Collect data needed to move VM datasets
+        facts['vm_facts_move_hypervisors'].append(
+            {'org': config['org'], 'host': host, 'source_hypervisor': hypervisor_host,
+             'destination_hypervisor': hypervisor_for_vm})
 
-if __name__ == "__main__":
-  main()
```

Comment:
Please reintroduce the function calls

Reply:
Why? This script does nothing else?

Reply:
You can't use this script in another script without conflicts, the `if __name__ == "__main__":` check insures the main function is only executed if the script is called directly.